aptos-move/block-executor/src/proptest_types/types.rs

```diff
@@ -325,15 +319,21 @@ pub(crate) struct TransactionGen<V: Into<Vec<u8>> + Arbitrary + Clone + Debug +
 #[derive(Clone, Debug)]
 pub(crate) struct MockIncarnation<K, E> {
     /// A vector of keys to be read during mock incarnation execution.
-    pub(crate) reads: Vec<K>,
+    /// bool indicates that the path may contain deltas, i.e. AggregatorV1 or DelayedFields.
+    pub(crate) reads: Vec<(K, bool)>,
     /// A vector of keys and corresponding values to be written during mock incarnation execution.
-    pub(crate) writes: Vec<(K, ValueType)>,
-    pub(crate) group_reads: Vec<(K, u32)>,
-    pub(crate) group_writes: Vec<(K, StateValueMetadata, HashMap<u32, ValueType>)>,
+    /// bool indicates that the path may contain deltas, i.e. AggregatorV1 or DelayedFields.
+    pub(crate) writes: Vec<(K, ValueType, bool)>,
+    pub(crate) group_reads: Vec<(K, u32, bool)>,
```

Comment:
Can we have struct instead of these tuples? Less changes in the future, and clearer when reading tests (structs can have constructors like `with_delayd_fields`, etc.? Also helps with reviewing so no need to cross-check this definition when reviewing other files.

Reply:
added a large TODO to also transform mock transaction generation into a builder pattern, and make things into more readable tuples. I think it makes sense to do that refactoring at the same time.